src/govuk/components/cookie-banner/template.test.js

```diff
@@ -0,0 +1,219 @@
+
+      const $heading = $('.govuk-cookie-banner__heading')
+      expect($heading.text()).toEqual('Cookies on this government service')
+    })
+
+    it('renders heading as escaped html when passed as text', () => {
+      const $ = render('cookie-banner', examples['heading html as text'])
+
+      const $heading = $('.govuk-cookie-banner__heading')
+      expect($heading.html().trim()).toEqual('Cookies on &lt;span&gt;my service&lt;/span&gt;')
+    })
+
+    it('renders heading html', () => {
+      const $ = render('cookie-banner', examples['heading html'])
+
+      const $heading = $('.govuk-cookie-banner__heading')
+      expect($heading.html().trim()).toEqual('Cookies on <span>my service</span>')
+    })
+
+    it('renders main content text', () => {
+      const $ = render('cookie-banner', examples.default)
+
+      const $content = $('.govuk-cookie-banner__content')
+      expect($content.text()).toEqual('We use analytics cookies to help understand how users use our service.')
+    })
+
+    it('renders main content html', () => {
+      const $ = render('cookie-banner', examples.html)
+
+      const $content = $('.govuk-cookie-banner__content')
+      expect($content.html().trim()).toEqual('<p class="govuk-body">We use cookies in <span>our service</span>.</p>')
+    })
+
+    it('renders classes', () => {
+      const $ = render('cookie-banner', examples.classes)
+
+      const $banner = $('.govuk-cookie-banner .govuk-cookie-banner__container')
+
+      expect($banner.hasClass('app-my-class')).toBeTruthy()
+    })
+
+    it('renders attributes', () => {
+      const $ = render('cookie-banner', examples.attributes)
+
+      const $banner = $('.govuk-cookie-banner .govuk-cookie-banner__container')
+
+      expect($banner.attr('data-attribute')).toEqual('my-value')
+    })
+  })
+
+  describe('role and aria attributes', () => {
+    it('has role=region by default', () => {
+      const $ = render('cookie-banner', examples.default)
+
+      const $component = $('.govuk-cookie-banner')
+      expect($component.attr('role')).toEqual('region')
+    })
+
+    it('has a default aria-label', () => {
+      const $ = render('cookie-banner', examples.default)
+
+      const $component = $('.govuk-cookie-banner')
+      expect($component.attr('aria-label')).toEqual('Cookie banner')
+    })
+
+    it('renders a custom aria label', () => {
+      const $ = render('cookie-banner', examples['custom aria label'])
+
+      const $component = $('.govuk-cookie-banner')
+
+      // expect($component.attr('role')).toEqual('region')
+      expect($component.attr('aria-label')).toEqual('Cookies on GOV.UK')
+    })
+  })
+
+  describe('confirmation banner', () => {
+    it('passes accessibility tests', async () => {
+      const $ = render('cookie-banner', examples['accepted confirmation banner'])
+
+      const results = await axe($.html())
+      expect(results).toHaveNoViolations()
+    })
+
+    it('set role attribute to alert', () => {
```

Comment:
This test sort of implies that there's some logic that means confirmation banners automatically get a role of `alert` – wonder if the distinction between question and confirmation banners is helpful as it's not really a distinction that exists in the code.

Reply:
I did wonder that, but the pattern of 'default' options that we normally use for our other components didn't really seem to fit here either. I've reworded the name of the test to `sets role attribute when role provided` which hopefully makes it clearer that it's not automatic?